src/ibei/models.py

```diff
@@ -151,34 +151,66 @@ def upper(self) -> astropy.units.Quantity:
         ----------
         .. [1] :cite:`10.1016/j.sse.2006.06.017`
         """
-        kT = self.temperature * astropy.constants.k_B
-
-        reduced_energy_bound = self.energy_bound / kT
-        reduced_chemical_potential = self.chemical_potential / kT
-
-        prefactor = (2 * np.pi * np.math.factorial(self.order) * kT**(self.order + 1)) / \
-            (astropy.constants.h**3 * astropy.constants.c**2)
-
-        expt = (reduced_chemical_potential - reduced_energy_bound).decompose()
+        expt = (self._reduced_chemical_potential - self._reduced_energy_bound).decompose()
```

Comment:
I might want to call `.decompose` on the quantities computed by `self._reduced_chemical_potential` and `self._reduced_energy_bound` so those outputs are unitless.